stock_picking_invoice_link/models/sale_order.py

```diff
@@ -1,26 +1,15 @@
 # Copyright 2013-15 Agile Business Group sagl (<http://www.agilebg.com>)
 # Copyright 2017 Jacques-Etienne Baudoux <je@bcim.be>
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
-
-from odoo import api, models
+from odoo import models
+from odoo.tools import float_compare
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    @api.multi
-    def invoice_line_create_vals(self, invoice_id, qty):
-        self.mapped("move_ids").filtered(
-            lambda x: x.state == "done"
-            and not x.invoice_line_id
-            and not x.location_dest_id.scrap_location
-            and x.location_dest_id.usage == "customer"
-        ).mapped("picking_id").write({"invoice_ids": [(4, invoice_id)]})
-        return super(SaleOrderLine, self).invoice_line_create_vals(invoice_id, qty)
-
-    @api.multi
-    def _prepare_invoice_line(self, qty):
-        vals = super(SaleOrderLine, self)._prepare_invoice_line(qty)
+    def x_prepare_invoice_line(self):
```

Comment:
I gues this method goes away in favor of the one below...

Reply:
Yes... of course!!!!